plugin/src/main/java/org/opensearch/ml/action/tasks/CancelBatchJobTransportAction.java

```diff
@@ -0,0 +1,226 @@
+        MLCancelBatchJobRequest mlCancelBatchJobRequest = MLCancelBatchJobRequest.fromActionRequest(request);
+        String taskId = mlCancelBatchJobRequest.getTaskId();
+        GetRequest getRequest = new GetRequest(ML_TASK_INDEX).id(taskId);
+
+        try (ThreadContext.StoredContext context = client.threadPool().getThreadContext().stashContext()) {
+            client.get(getRequest, ActionListener.runBefore(ActionListener.wrap(r -> {
+                log.debug("Completed Get Task Request, id:{}", taskId);
+
+                if (r != null && r.isExists()) {
+                    try (XContentParser parser = createXContentParserFromRegistry(xContentRegistry, r.getSourceAsBytesRef())) {
+                        ensureExpectedToken(XContentParser.Token.START_OBJECT, parser.nextToken(), parser);
+                        MLTask mlTask = MLTask.parse(parser);
+
+                        // check if function is remote and task is of type batch prediction
+                        if (mlTask.getTaskType() == MLTaskType.BATCH_PREDICTION && mlTask.getFunctionName() == FunctionName.REMOTE) {
+                            processRemoteBatchPrediction(mlTask, actionListener);
+                        } else {
+                            actionListener
+                                .onFailure(new IllegalArgumentException("The task ID you provided does not have any associated batch job"));
+                        }
+                    } catch (Exception e) {
+                        log.error("Failed to parse ml task " + r.getId(), e);
+                        actionListener.onFailure(e);
+                    }
+                } else {
+                    actionListener.onFailure(new OpenSearchStatusException("Fail to find task", RestStatus.NOT_FOUND));
+                }
+            }, e -> {
+                if (e instanceof IndexNotFoundException) {
+                    actionListener.onFailure(new MLResourceNotFoundException("Fail to find task"));
+                } else {
+                    log.error("Failed to get ML task " + taskId, e);
+                    actionListener.onFailure(e);
+                }
+            }), () -> context.restore()));
+        } catch (Exception e) {
+            log.error("Failed to get ML task " + taskId, e);
+            actionListener.onFailure(e);
+        }
+    }
+
+    private void processRemoteBatchPrediction(MLTask mlTask, ActionListener<MLCancelBatchJobResponse> actionListener) {
+        Map<String, Object> remoteJob = mlTask.getRemoteJob();
+
+        Map<String, String> parameters = new HashMap<>();
+        for (Map.Entry<String, ?> entry : remoteJob.entrySet()) {
+            if (entry.getValue() instanceof String) {
+                parameters.put(entry.getKey(), (String) entry.getValue());
+            } else {
+                log.debug("Value for key " + entry.getKey() + " is not a String");
+            }
+        }
```

Comment:
Yeah it seems easier when remote job is Map<String, String> while creating itself. Will make that change to avoid all this confusion

Reply:
When converting to Map<String, String>, the values that are not strings will be lost. User may not be able to see complete response from remote job. I guess it is safe to use Map<String, Object> so that all key-value pairs from remote service are retained